### PR TITLE
docs - PL/Container updates for 1.4.0

### PR DESCRIPTION
**Pull request by mkiyama**
PL/Python updates
--support for plpy.Error, plpy.Fatal returning nested function information
--support for quoting functions plpy.quote_literal, plpy.quote_nullable, plpy.quote_ident
--returning Python unicode text as text in server encoding.

This will be backported to 5X_STABLE

Link to HTML format on the GPDB docs review site.
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/extensions/pl_container.html#topic_ehl_r3q_dw

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/extensions/pl_container.xml [hunk 1]
```diff
@@ -706,6 +695,28 @@ $$ LANGUAGE plcontainer;</codeblock></p>
...
           <codeph>plpy.subtransaction()</codeph>.</li>
       </ul>
+      <p>If an error of level ERROR or FATAL is raised in a nested Python function call, the message
+        includes the list of enclosing functions.</p>
+      <p>The Python language container supports these string quoting functions that are useful when
+        constructing ad-hoc queries. <ul id="ul_jwf_nd2_kfb">
+          <li><codeph>plpy.quote_literal(string)</codeph> - Returns the string quoted to be used as
+            a string literal in an SQL statement string. Embedded single-quotes and backslashes are
+            properly doubled. <codeph>quote_literal</codeph> returns null on null input (empty
+            input). If the argument might be null, <codeph>quote_nullable</codeph> might be more
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/extensions/pl_container.xml]: you might consider adding a `()` after the references to `quote_literal` and `quote_nullable` to make it obvious you are referring to a function.

File: gpdb-doc/dita/ref_guide/extensions/pl_container.xml [hunk 2]
```diff
@@ -706,6 +695,28 @@ $$ LANGUAGE plcontainer;</codeblock></p>
           documentation for additional information about
           <codeph>plpy.subtransaction()</codeph>.</li>
       </ul>
+      <p>If an error of level ERROR or FATAL is raised in a nested Python function call, the message
+        includes the list of enclosing functions.</p>
+      <p>The Python language container supports these string quoting functions that are useful when
+        constructing ad-hoc queries. <ul id="ul_jwf_nd2_kfb">
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/extensions/pl_container.xml]: are these new functions also directly supported by pl/python?  i.e. do we need to add them to that page as well?

Comment (mkiyama) [reply gpdb-doc/dita/ref_guide/extensions/pl_container.xml]: The section "About PL/Container Running PL/Python" is just a overview of PL/Python support in PL/Container, mostly the plpy module support with a link to the GPDB PL/Python topic at the end.

I didn't have much time for other than that.

The plpy quoting functions are implemented in the plpy module in PL/Container. 

I have a question out to dev if the functions are implemented in GPDB PL/Python. 
I kind of doubt it for 5X_STABLE, I haven't see a lot of work on PL/Python.

FYI - on master, PL/Python is currently up to postgres 9.3.

**Comment by mkiyama** [comment]
@zhangh43 
Have you had a chance to review this PR?

thanks

File: gpdb-doc/dita/ref_guide/extensions/pl_container.xml [hunk 3]
```diff
@@ -706,6 +695,28 @@ $$ LANGUAGE plcontainer;</codeblock></p>
...
+          <li><codeph>plpy.quote_literal(string)</codeph> - Returns the string quoted to be used as
+            a string literal in an SQL statement string. Embedded single-quotes and backslashes are
+            properly doubled. <codeph>quote_literal()</codeph> returns null on null input (empty
+            input). If the argument might be null, <codeph>quote_nullable()</codeph> might be more
+            appropriate.</li>
+          <li><codeph>plpy.quote_nullable(string)</codeph> Returns the string quoted to be used as a
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/extensions/pl_container.xml]: Add dashes after the function names in the remaining list items (to be consistent with the previous item, and the other list of R functions).

**Pull request merged**